adempiere/adempiere
# #213 Add ability to automatically synchronize table and column changes with the database.

Fixes #213.

Enable database autosync. For example, a change to a system element could be made that changes a column name and that change would be made in all affected tables in the database without the need to visit each column and run the ColumnSync process.

Also included is the filling of mandatory columns with default data if the column is added to an existing table.  A typical example is the addition of a mandatory yes/no field with a default of N.

There are some  limitations:
- check constraints are not renamed. 
- does not update software models or references to the table/column in views or queries.

This enhancement is really aimed at making it easier to create tables and columns on the fly during development.

The auto sync is configured to run as default but a system configurator entry could be added to turn it off.

Test as follows:

- Open your database admin tool to watch the changes.
- Log into ADempiereas system or superuser in sysadmin role
- Create a test element for a dummy column, say 'Test_Element'
- Create a new test table called 'Test_Table'. Save the record. Note the columns, especially the key column, are created. Do not click the Synchronize button.
- Check in the database admin tool that the table has been created and all the default columns exist.
- Add a column to the table with your test element. Save the record.
- In the database admin tool note that the column has been added to the database table.
- In ADempiere, change the table name. Note that the key column name has also changed.
- Check in the database admin tool that the table name has changed and that the name of the key column and associated constraint has also changed. A table with the original name should not exist.
